cpp/src/arrow/python/iterators.h

```diff
@@ -84,18 +84,29 @@ inline Status VisitSequenceGeneric(PyObject* obj, VisitorFunc&& func) {
   return Status::OK();
 }
 
-// Visit sequence with no null mask
+// Visit sequence with no null mask or offset
 template <class VisitorFunc>
 inline Status VisitSequence(PyObject* obj, VisitorFunc&& func) {
```

Comment:
How about always passing `offset`? That will simplify the overloads a bit (this should not be considered a public API).